src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs

```diff
@@ -119,26 +130,54 @@ private void Process_Exited(object sender, EventArgs e)
         {
             lock (syncObject)
             {
-                vstestConsoleExited = true;
+                ShutdownProcess();
+
                 this.ProcessExited?.Invoke(sender, e);
             }
         }
 
-        private string[] BuildArguments(ConsoleParameters parameters)
+        private void Process_ErrorDataReceived(object sender, DataReceivedEventArgs e)
         {
-            var args = new List<string>();
+            if (e.Data != null)
+            {
+                EqtTrace.Error(e.Data);
+            }
+        }
+
+        private void Process_OutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            if (e.Data != null)
+            {
+                EqtTrace.Verbose(e.Data);
+            }
+        }
 
-            // Start Vstest.console.exe with args: --parentProcessId|/parentprocessid:<ppid> --port|/port:<port>
-            args.Add(string.Format(CultureInfo.InvariantCulture, PARENT_PROCESSID_ARGUMENT, parameters.ParentProcessId));
-            args.Add(string.Format(CultureInfo.InvariantCulture, PORT_ARGUMENT, parameters.PortNumber));
+        private string[] BuildArguments(ConsoleParameters parameters)
+        {
+            var args = new List<string>
+            {
+                // Start Vstest.console with args: --parentProcessId|/parentprocessid:<ppid> --port|/port:<port>
+                string.Format(CultureInfo.InvariantCulture, PARENT_PROCESSID_ARGUMENT, parameters.ParentProcessId),
+                string.Format(CultureInfo.InvariantCulture, PORT_ARGUMENT, parameters.PortNumber)
+            };
 
-            if(!string.IsNullOrEmpty(parameters.LogFilePath))
+            if (!string.IsNullOrEmpty(parameters.LogFilePath))
             {
                 // Extra args: --diag|/diag:<PathToLogFile>;tracelevel=<tracelevel>
                 args.Add(string.Format(CultureInfo.InvariantCulture, DIAG_ARGUMENT, parameters.LogFilePath, parameters.TraceLevel));
             }
 
+            if (isNetCoreRunner)
+            {
+                args.Insert(0, vstestConsolePath);
+            }
+
             return args.ToArray();
         }
+
+        private string GetConsoleRunner()
+        {
+            return isNetCoreRunner ? DOTNET_EXECUTABLE : vstestConsolePath;
```

Comment:
>DOTNET_EXECUTABLE [](start = 37, length = 17)

The way we find path for dotnet.exe on different platforms is shared at [https://github.com/Microsoft/vstest/blob/7b6248203164f8ea821f6795632bd22e0b69afb0/src/Microsoft.TestPlatform.CoreUtilities/Helpers/DotnetHostHelper.cs#L43](https://github.com/Microsoft/vstest/blob/7b6248203164f8ea821f6795632bd22e0b69afb0/src/Microsoft.TestPlatform.CoreUtilities/Helpers/DotnetHostHelper.cs#L43)

Here it seems you are assuming that dotnet is installed on machine, & is added to path variables. Can you please handle scenarios where it might not be the case